app/controllers/users.controller.js

```diff
@@ -356,4 +359,95 @@ exports.getUserFromToken = function (req, res) {
             return res.status(200).json(user);
         });
     }
-};
\ No newline at end of file
+};
+
+
+
+exports.updateFirstName = function (req, res) {
+    var firstName = (req.body.firstName) ? req.body.firstName.trim() : '';
+    var user = AuthUtils.getUserFromToken(req);
+    User.findById(user._id, function(err, user) {
```

Comment:
Make sure to check that the user is not null here. If it is return a 401, 'unauthorized.'

Reply:
```
    if (!user) {
        res.status(401).send('unauthorized');
    } else {
        User.findById(user._id, function(err, user) {
            if (err) {
                res.status(500).send({successful: false, text: err.message});
            } else if (!user) {
                res.status(400).send({successful: false, text: 'Could not find user.'})
            } else {
                user.firstName = firstName;
                user.save(function(err) {
                    if (err) {
                        res.status(500).send({successful: false, text: err.message});
                    } else {
                        res.status(200).send({successful: true, text: 'Your first name has been '
                            + 'successfully changed to ' + firstName + ' !'});
                    }
                });
            }
        });
    }
```